src/mmrelay/matrix/relay.py

```
@@ -0,0 +1,429 @@
+                delay = _retry_backoff_delay(attempt, base_delay, max_delay)
+                jitter = rng.uniform(0, delay * 0.1)
+                total_delay = delay + jitter
+                facade.logger.warning(
+                    f"Timeout sending to Matrix room {room_id} (attempt {attempt + 1}/{max_retries + 1}), "
+                    f"retrying in {total_delay:.1f}s..."
+                )
+                await asyncio.sleep(total_delay)
+            else:
+                facade.logger.exception(
+                    f"Timeout sending message to Matrix room {room_id} after {max_retries + 1} attempts"
+                )
```

Ruff (0.15.9)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)